pdp_util/raster.py

```diff
@@ -238,22 +239,80 @@ def __call__(self, environ, start_response):
         return response_200(start_response, content)
 
 
-def build_orca_url(handlers, thredds_root, req):
+def build_orca_url(handlers, orca_root, req):
     """orca is the OPeNDAP Request Compiler Application which pulls apart large OPeNDAP requests
     to THREDDS into bite-sized chunks and then reasemmbles them for the user.
 
-    Orca is available through a url with the format:
-    [thredds_root]/[filepath]:[variable][time_start:time_end][lat_start:lat_end][lon_start:lon_end]
+    Orca is available through a url with one of the formats:
+    1. [orca_root]/?filepath=[filepath]
+    2. [orca_root]/?filepath=[filepath]&targets=time[time_start:time_end],lat[lat_start:lat_end],lon[lon_start:lon_end],[variable][time_start:time_end][lat_start:lat_end][lon_start:lon_end]
 
     where the [filepath] can be attained by the mapping of handler url to handler file from a config dict
     """
     filename = None
     for handler in handlers:
+        # print("handler url: {} path_info: {}".format(handler["url"], req.path_info[:-3]))
         if handler["url"] == req.path_info[:-3]:
             filename = handler["file"]
             break
+        elif handler["url"] == req.path_info[:-3].strip("/."):
+            filename = handler["file"].strip("/,")
+            break
+
+    if req.query_string == "":
+        return f"{orca_root}/?filepath={filename}"
+    else:
+        dims = get_target_dims(req.query_string.rstrip("&"))
+        return f"{orca_root}/?filepath={filename}&targets={dims}{req.query_string.rstrip('&')}&outfile={req.path_info.strip('/.')}"
+
+
+def get_target_dims(var):
+    """Adds the dimensions with the bounds matching the data variable to the targets
+    for orca data requests."""
+    try:
+        bounds = var[var.index("[") :]
+    except ValueError:  # Get full range of data variable
+        return "time,lat,lon,"
+
+    split_bounds = [bound + "]" for bound in bounds.split("]")][:-1]
```

Comment:
I'm having trouble figuring out exactly what is in `split_bounds` and what the ultimate value computed from it (`target_dims`) is. 

- Would this computation be simpler `bounds` were computed as `var[var.index("[") : -1]`? 
- Adding info in the docstring specifying the return value more exactly would help, plus maybe some intermediate comments in the body. 
- And/or tests of this function.